core/internal/api/credentials_test.go

```diff
@@ -29,14 +35,271 @@ func TestNewAPIKeyCredentialProvider(t *testing.T) {
 
 func TestNewAPIKeyCredentialProvider_NoAPIKey(t *testing.T) {
 	settings := wbsettings.From(&spb.Settings{})
-	_, err := api.NewCredentialProvider(settings)
+	_, err := api.NewCredentialProvider(settings, observability.NewNoOpLogger().Logger)
 	assert.Error(t, err)
 }
 
-func TestNewAccessTokenCredentialProvider(t *testing.T) {
+func authServer(token string, expiresIn time.Duration) *apitest.RecordingServer {
+	handler := func(w http.ResponseWriter, req *http.Request) {
+		response := map[string]interface{}{
+			"access_token": token,
+			"expires_in":   expiresIn.Seconds(),
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+
+		if err := json.NewEncoder(w).Encode(response); err != nil {
+			http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+			return
+		}
+	}
+	server := apitest.NewRecordingServer(apitest.WithHandlerFunc(handler))
+	return server
+}
+
+func TestNewOAuth2CredentialProvider(t *testing.T) {
+	// create identity token file
+	tokenFile, err := os.CreateTemp("", "jwt.txt")
```

Comment:
For temporary directories in tests, please use https://pkg.go.dev/testing#T.TempDir